internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -124,11 +126,18 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	cluster := &fleet.Cluster{}
 	err := r.Get(ctx, req.NamespacedName, cluster)
-	if apierrors.IsNotFound(err) {
-		metrics.ClusterCollector.Delete(req.Name, req.Namespace)
-		return ctrl.Result{}, nil
-	} else if err != nil {
-		return ctrl.Result{}, err
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !cluster.DeletionTimestamp.IsZero() {
+		return r.handleDelete(ctx, cluster)
```

Comment:
I think this requires an additional RBAC permission for deleting the namespaces.
```
//+kubebuilder:rbac:groups="",resources=namespaces,verbs=delete
```